testsuite/cluster-test/src/atomic_histogram.rs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+#![forbid(unsafe_code)]
+
+use crate::tx_emitter::TxStats;
+use std::sync::atomic::{AtomicU64, Ordering};
+
+const HISTOGRAM_CAPACITY: usize = 2048;
+const HISTOGRAM_WIDTH_MILLIS: u64 = 50;
+
+pub struct AtomicHistogram {
+    buckets: Vec<AtomicU64>,
+}
+
+impl AtomicHistogram {
+    pub fn default() -> Self {
+        let mut buf = Vec::with_capacity(HISTOGRAM_CAPACITY);
+        for _i in 0..HISTOGRAM_CAPACITY {
+            buf.push(AtomicU64::new(0));
+        }
+        Self { buckets: buf }
+    }
+
+    pub fn to_vec(&self) -> Vec<u64> {
+        let mut buf = Vec::with_capacity(HISTOGRAM_CAPACITY);
+        for i in 0..HISTOGRAM_CAPACITY {
+            buf.push(self.buckets[i].load(Ordering::Relaxed));
+        }
+
+        buf
+    }
+
+    fn get_bucket_num(&self, latency: u64) -> u64 {
+        let bucket_num = latency / HISTOGRAM_WIDTH_MILLIS;
+        if bucket_num >= HISTOGRAM_CAPACITY as u64 - 2 {
+            return HISTOGRAM_CAPACITY as u64 - 1;
+        }
+        bucket_num
+    }
+
+    pub fn record_latency_in_bucket(&self, latency: u64, num_requests: u64) {
```

Comment:
You may want to rename `latency` -> `latency_ms`
Also `record_latency_in_bucket` -> `record_latency_ms`